src/java/org/apache/cassandra/repair/RepairJob.java

```diff
@@ -164,19 +198,59 @@ public void onFailure(Throwable t)
             return;
         }
 
+        Future<Void> accordRepair;
+        if (doAccordRepair)
+        {
+            accordRepair = paxosRepair.flatMap(unused -> {
+                logger.info("{} {}.{} starting accord repair", session.previewKind.logPrefix(session.getId()), desc.keyspace, desc.columnFamily);
+                IPartitioner partitioner = desc.ranges.iterator().next().left.getPartitioner();
```

Comment:
```suggestion
                IPartitioner partitioner = metadata.getPartitioner();
```

rather than walking ranges, we know the table why not pull there?